broker/src/main/java/io/camunda/zeebe/broker/exporter/metrics/MetricsExporter.java

```diff
@@ -131,33 +129,28 @@ private void handleProcessInstanceRecord(
 
     if (currentIntent == ProcessInstanceIntent.ELEMENT_ACTIVATING
         && isProcessInstanceRecord(record)) {
-      storeProcessInstanceCreation(record.getTimestamp(), recordKey);
+      processInstanceCache.store(recordKey, record.getTimestamp());
     } else if (currentIntent == ProcessInstanceIntent.ELEMENT_COMPLETED
         && isProcessInstanceRecord(record)) {
-      final var creationTime = processInstanceKeyToCreationTimeMap.remove(recordKey);
+      final var creationTime = processInstanceCache.remove(recordKey);
```

Comment:
:thought_balloon: When the key is not in the cache, this will update the latency as `completedTime - (-1)`, which would skew the metrics. This is not introduced in this PR though. But would be nice to handle this case.

Reply:
Any ideas? We could simply not report it I suppose. Not sure why we were doing it before.

Reply:
I first though simply not reporting it would ok. But this also skews the result because we might not report any long running instances. May be cap it to threshold like a minute. May be use the default TIME_TO_LIVE as it is the cleanup interval. It is still not accurate, but better than us assuming it is all good because long lived instances are never included in the metric.

Reply:
Capping is probably the "best" solution, agreed

Reply:
The "null" value is now customizable instead of fixed at -1, so the exporter uses the TTL for missing values from now on :+1: